File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/FrmInicio.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports CapaNegocio
Imports System.IO

Public Partial Class FrmInicio
	Inherits Form
	Private empleados As Empleados
	Private transporte As Transporte
	Private proveedores As Proveedores
	Private categorias As Categorias
	Private productos As Productos
	Private ordenes As Ordenes
	Private reportes As FrmReportes

	Public Sub New()
		Dim mensaje As [String] = NConexion.ChequearConexion()
		If mensaje = "Y" Then
			InitializeComponent()
			Iniciar()
		Else
			MessageBox.Show("El Sistema de Compras no se puede iniciar ya que existió un error al conectarse con la Base de Datos. " & mensaje, "Error al iniciar el Sistema de Compras", MessageBoxButtons.OK, MessageBoxIcon.[Error])
			Me.Dispose()
		End If
	End Sub

	Public Sub Iniciar()
		' ordenes = new Ordenes();
		' this.panelContenedor.Controls.Clear();
		' this.panelContenedor.Controls.Add(ordenes);
	End Sub

	'private void buttonEmpleados_Click(object sender, EventArgs e)
'        {
'            if (empleados == null)
'            {
'                empleados = new Empleados();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(empleados);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(empleados);
'            }
'        }
'
'        private void buttonTransporte_Click(object sender, EventArgs e)
'        {
'            if (transporte == null)
'            {
'                transporte = new Transporte();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(transporte);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(transporte);
'            }
'        }
'
'        private void buttonClientes_Click(object sender, EventArgs e)
'        {
'            if (proveedores == null)
'            {
'                proveedores = new Proveedores();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(proveedores);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(proveedores);
'            }
'        }
'
'        private void buttonCategorias_Click(object sender, EventArgs e)
'        {
'            if (categorias == null)
'            {
'                categorias = new Categorias();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(categorias);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(categorias);
'            }
'        }
'
'        private void buttonProductos_Click(object sender, EventArgs e)
'        {
'            if (productos == null)
'            {
'                productos = new Productos();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(productos);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(productos);
'            }
'        }
'
'        private void buttonPedidos_Click(object sender, EventArgs e)
'        {
'            if (ordenes == null)
'            {
'                ordenes = new Ordenes();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(ordenes);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(ordenes);
'            }
'        }
'
'        private void buttonReportes_Click(object sender, EventArgs e)
'        {
'            try
'            {
'                FrmReportes administracionReportes = new FrmReportes();
'                administracionReportes.ShowDialog();
'            }
'
'            catch (Exception ex)
'            {
'                MensajeError(ex.Message);
'            }
'        }


	Public Sub MensajeError(mensaje As String)
		MessageBox.Show(mensaje, "Error", MessageBoxButtons.OK, MessageBoxIcon.[Error])
	End Sub


	Private Sub salirToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Me.Close()
	End Sub


	Private Sub salirToolStripMenuItem1_Click(sender As Object, e As EventArgs)
		Me.Close()
	End Sub

	Private Sub ordenesDeCompraToolStripMenuItem_Click(sender As Object, e As EventArgs)
		' if (ordenes == null)
'            {
'                ordenes = new Ordenes();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(ordenes);
'
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(ordenes);
'            }


		ordenes = New Ordenes()
		ordenes.Show()

	End Sub

	Private Sub productoToolStripMenuItem_Click(sender As Object, e As EventArgs)
		'if (productos == null)
'            {
'                productos = new Productos();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(productos);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(productos);
'            }


		productos = New Productos()
		productos.Show()
	End Sub

	Private Sub proveedoresToolStripMenuItem_Click(sender As Object, e As EventArgs)
		'if (proveedores == null)
'            {
'                proveedores = new Proveedores();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(proveedores);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(proveedores);
'            }


		proveedores = New Proveedores()
		proveedores.Show()
	End Sub

	Private Sub categoriasToolStripMenuItem_Click(sender As Object, e As EventArgs)
		' if (categorias == null)
'             {
'                 categorias = new Categorias();
'                 this.panelContenedor.Controls.Clear();
'                 this.panelContenedor.Controls.Add(categorias);
'             }
'             else
'             {
'                 this.panelContenedor.Controls.Clear();
'                 this.panelContenedor.Controls.Add(categorias);
'             }


		categorias = New Categorias()
		categorias.Show()

	End Sub

	Private Sub transporteToolStripMenuItem_Click(sender As Object, e As EventArgs)
		' if (transporte == null)
'             {
'                 transporte = new Transporte();
'                 this.panelContenedor.Controls.Clear();
'                 this.panelContenedor.Controls.Add(transporte);
'             }
'             else
'             {
'                 this.panelContenedor.Controls.Clear();
'                 this.panelContenedor.Controls.Add(transporte);
'             }

		transporte = New Transporte()
		transporte.Show()
	End Sub

	Private Sub usuariosToolStripMenuItem_Click(sender As Object, e As EventArgs)
		'if (empleados == null)
'            {
'                empleados = new Empleados();
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(empleados);
'            }
'            else
'            {
'                this.panelContenedor.Controls.Clear();
'                this.panelContenedor.Controls.Add(empleados);
'            }


		empleados = New Empleados()
		empleados.Show()

	End Sub

	Private Sub reportesToolStripMenuItem_Click(sender As Object, e As EventArgs)
		'try
'            {
'                FrmReportes administracionReportes = new FrmReportes();
'                administracionReportes.ShowDialog();
'            }
'
'            catch (Exception ex)
'            {
'                MensajeError(ex.Message);
'            }


		reportes = New FrmReportes()
		reportes.Show()
	End Sub


	Private Sub reportesToolStripMenuItem1_Click(sender As Object, e As EventArgs)
		Try
			Dim administracionReportes As New FrmReportes()
			administracionReportes.ShowDialog()

		Catch ex As Exception
			MensajeError(ex.Message)
		End Try
	End Sub


End Class

File: Programacion Modulo Compras/SAE-COMPRAS/CapaPresentacion/FrmInicio.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using CapaNegocio;
using System.IO;

namespace CapaPresentacion
{
    public partial class FrmInicio : Form
    {
        Proveedores proveedores;
        Categorias categorias;
        Productos productos;
        Ordenes ordenes;
        FrmReportes reportes;
        Reporte1 reporte1;

        public FrmInicio()
        {
            String mensaje = NConexion.ChequearConexion();
            if (mensaje == "Y")
            {
                InitializeComponent();

            }
            else
            {
                MessageBox.Show("El Sistema de Compras no se puede iniciar ya que existió un error al conectarse con la Base de Datos. " + mensaje,
                  "Error al iniciar el Sistema de Compras", MessageBoxButtons.OK, MessageBoxIcon.Error);
                this.Dispose();
            }
        }


        public void MensajeError(string mensaje)
        {
            MessageBox.Show(mensaje, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
        }


        private void salirToolStripMenuItem_Click(object sender, EventArgs e)
        {
            this.Close();
        }


        private void salirToolStripMenuItem1_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void ordenesDeCompraToolStripMenuItem_Click(object sender, EventArgs e)
        {

            ordenes = new Ordenes();
            ordenes.Show();

        }

        private void productoToolStripMenuItem_Click(object sender, EventArgs e)
        {

            productos = new Productos();
            productos.Show();
        }

        private void proveedoresToolStripMenuItem_Click(object sender, EventArgs e)
        {

            proveedores = new Proveedores();
            proveedores.Show();
        }

        private void categoriasToolStripMenuItem_Click(object sender, EventArgs e)
        {

            categorias = new Categorias();
            categorias.Show();

        }

        private void reportesToolStripMenuItem_Click(object sender, EventArgs e)
        {

            reportes = new FrmReportes();
            reportes.Show();
        }


        private void reportesToolStripMenuItem1_Click(object sender, EventArgs e)
        {
           
        }

        private void ordenesDeCompraToolStripMenuItem1_Click(object sender, EventArgs e)
        {
            ordenes = new Ordenes();
            ordenes.Show();
        }

        private void agregarNuevoProductoToolStripMenuItem_Click(object sender, EventArgs e)
        {
        }

        private void menuStrip1_ItemClicked(object sender, ToolStripItemClickedEventArgs e)
        {

        }

        private void ordenesPorProveedorToolStripMenuItem_Click(object sender, EventArgs e)
        {
            reporte1 = new Reporte1();
            reporte1.Show();
        }

        private void agregarNuevoSocioToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Reporte2 agregar = new Reporte2();
            agregar.Show();
        }

        private void sociosToolStripMenuItem_Click(object sender, EventArgs e)
        {
            proveedores = new Proveedores();
            proveedores.Show();
        }

        private void procesosToolStripMenuItem_Click(object sender, EventArgs e)
        {
         
        }

        private void productosToolStripMenuItem_Click(object sender, EventArgs e)
        {
            productos = new Productos();
            productos.Show();
        }

        private void historialProveedoresToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Reporte2 agregar = new Reporte2();
            agregar.Show();
        }

        private void panelContenedor_Paint(object sender, PaintEventArgs e)
        {

        }

        private void reportesPorSocioYProductoToolStripMenuItem_Click(object sender, EventArgs e)
        {
            try
            {
                FrmReportes administracionReportes = new FrmReportes();
                administracionReportes.ShowDialog();
            }

            catch (Exception ex)
            {
                MensajeError(ex.Message);
            }
        }

        private void proveedoresYProductosToolStripMenuItem_Click(object sender, EventArgs e)
        {
        }
    }
    }


